commands/operator-sdk/cmd/generate/crd.go

```diff
@@ -0,0 +1,116 @@
+// limitations under the License.
+
+package generate
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	cmdError "github.com/operator-framework/operator-sdk/commands/operator-sdk/error"
+	"github.com/operator-framework/operator-sdk/pkg/generator"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	apiVersion string
+	kind       string
+)
+
+const (
+	goDir        = "GOPATH"
+	deployCrdDir = "deploy"
+)
+
+func NewGenerateCrdCmd() *cobra.Command {
+	crdCmd := &cobra.Command{
+		Use:   "crd",
+		Short: "Generates a custom resource definition (CRD) and the custom resource (CR) files",
+		Long: `The operator-sdk generate command will create a custom resource definition (CRD) and the custom resource (CR) files for the specified api-version and kind.
+
+Generated CRD filename: <project-name>/deploy/<kind>_crd.yaml
+Generated CR  filename: <project-name>/deploy/<kind>_cr.yaml
+
+	<project-name>/deploy path must already exist
+	--api-version and --kind are required flags to generate the new operator application.
+`,
+		Run: crdFunc,
+	}
+	crdCmd.Flags().StringVar(&apiVersion, "api-version", "", "Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
+	crdCmd.MarkFlagRequired("api-version")
+	crdCmd.Flags().StringVar(&kind, "kind", "", "Kubernetes CustomResourceDefintion kind. (e.g AppService)")
+	crdCmd.MarkFlagRequired("kind")
+	return crdCmd
+}
+
+func crdFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 0 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, errors.New("crd command doesn't accept any arguments."))
+	}
+	verifyCrdFlags()
+	verifyCrdDeployPath()
+
+	fmt.Fprintln(os.Stdout, "Generating custom resource definition (CRD) file")
+
+	// generate CRD file
+	wd, err := os.Getwd()
+	if err != nil {
+		cmdError.ExitWithError(cmdError.ExitError, err)
+	}
+	if err := generator.RenderDeployCrdFiles(filepath.Join(wd, deployCrdDir), apiVersion, kind); err != nil {
+		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to generate CRD and CR files: (%v)", err))
+	}
+}
+
+func verifyCrdFlags() {
+	if len(apiVersion) == 0 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, errors.New("--api-version must not have empty value"))
+	}
+	if len(kind) == 0 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, errors.New("--kind must not have empty value"))
+	}
+	kindFirstLetter := string(kind[0])
+	if kindFirstLetter != strings.ToUpper(kindFirstLetter) {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, errors.New("--kind must start with an uppercase letter"))
+	}
+	if strings.Count(apiVersion, "/") != 1 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("api-version has wrong format (%v); format must be $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)", apiVersion))
+	}
+}
+
+// verifyCrdDeployPath checks if the path <project-name>/deploy sub-directory is exists, and that is rooted under $GOPATH
+func verifyCrdDeployPath() {
+	// check if $GOPATH env exists
```

Comment:
While re-looking at the code, I would like to change one more thing about the generation as well

I think an ansible operator new, should not have to have Gopath set up. I think this would force users to have that, and I think we should avoid this. 

@johnkim76 I wonder if we should just check that the current working dir has a `deploy` directory instead?

Reply:
ok.. I'll remove the check for the Gopath

Reply:
Good catch Shawn. I agree we shouldn't depend on Go being configured.